packages/backend/src/managers/catalogManager.ts

```diff
@@ -289,9 +289,14 @@ export class CatalogManager extends Publisher<ApplicationCatalog> implements Dis
           result = res;
           break;
         }
-        case 'tools':
-          result = result.filter(r => values.includes(r.backend ?? ''));
+        case 'tools': {
+          let res: Recipe[] = [];
+          for (const value of values) {
+            res = [...res, ...result.filter(r => r.backends?.includes(value))];
+          }
+          result = res;
```

Comment:
question: I don't understand what this code is doing? can we add a comment? filtering result? shallow copy?